test/integration/webhook/v1alpha1/workload_test.go

```diff
@@ -70,3 +70,76 @@ var _ = ginkgo.Describe("Workload defaulting webhook", func() {
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Workload validating webhook", func() {
+	ginkgo.Context("When creating a Workload", func() {
+		ginkgo.It("Should validate spec.podSets.count", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload2", ns.Name).Obj()
+			workload.Spec.PodSets = []v1alpha1.PodSet{
+				{
+					Count: -1,
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "test",
+								Image: "fake-image",
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, workload)
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(err.Error()).Should(gomega.ContainSubstring("must be greater than 0"))
+		})
+
+		ginkgo.It("Should validate spec.priorityClassName", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload3", ns.Name).Obj()
+			workload.Spec.PriorityClassName = "invalid_class"
+			err := k8sClient.Create(ctx, workload)
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(err.Error()).Should(gomega.ContainSubstring("a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters"))
+		})
+	})
+
+	ginkgo.Context("When updating a Workload", func() {
+		ginkgo.It("Should validate spec.podSet.count", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload3", ns.Name).Obj()
+			workload.Spec.PodSets = []v1alpha1.PodSet{
+				{
+					Count: 2,
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "test",
+								Image: "fake-image",
+							},
+						},
+					},
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, workload)).Should(gomega.Succeed())
+
+			ginkgo.By("Updating the Workload")
+			workload.Spec.PodSets[0].Count = -1
+			err := k8sClient.Update(ctx, workload)
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(err.Error()).Should(gomega.ContainSubstring("must be greater than 0"))
+		})
+
+		ginkgo.It("Should validate spec.priorityClassName", func() {
```

Comment:
I would remove this test.

We need to ensure that the webhook validates something during create/update, and leave the rest to unit tests.